examples/react/next-app-router/app/page.tsx

```diff
@@ -0,0 +1,50 @@
+'use client';
```

Comment:
Whichever component that has `<InstantSearch>` at its root yes, but they're free to have a server component in `page.js` which also contains their search component

Reply:
Do you get a clear error if it's accidentally a server component? if not we should give some warnign/error

Reply:
They would as `<InstantSearch>` calls hooks, but probably not the clearest.
We'll have to test a lot of scenarios yes 😅

Reply:
The stack trace might not be clear enough, but the message explicitly say "Please add 'use client'" and the stack trace finishes at the `page.tsx` (or whatever page in which they add InstantSearch).

Reply:
Tried adding `'use client'` in `react-instantsearch/index.ts` but it throws an error that you can't do `export *` in a client boundary.
Anyway I've seen that Apollo and other libs have users place `'use client'` themselves, and I think it's probably better that they explicitly add their boundaries themselves rather than it being hidden in the library which can be confusing.